lib/text/mp4_ttml_parser.js

```diff
@@ -80,67 +82,117 @@ shaka.text.Mp4TtmlParser = class {
   parseMedia(data, time, uri) {
     const Mp4Parser = shaka.util.Mp4Parser;
 
-    let sawMDAT = false;
     let payload = [];
+    let defaultSampleSize = null;
+
+    /** @type {!Array<Uint8Array>} */
+    const mdat = [];
```

Comment:
I would call this variable `mdats`, to make it clearer that it is an array that contains multiple mdat sub-arrays, instead of a single mdat that we are concatenating data on to (like how we do with the `payload`).